src/main/kotlin/org/rust/lang/core/psi/RsDefaultValueBuilder.kt

```diff
@@ -19,32 +20,63 @@ class RsDefaultValueBuilder(
     private val defaultValue: RsExpr
         get() = psiFactory.createExpression("()")
 
-    fun buildFor(ty: Ty): RsExpr {
+    private fun buildForSmartPtr(ty: TyAdt, bindings: HashSet<RsPatBinding>): RsExpr {
+        val item = ty.item
+        val name = ty.item.name!!
+        val parameter = ty.typeParameterValues[item.typeParameters[0]]!!
+        return psiFactory.createAssocFunctionCall(name, "new", listOf(buildFor(parameter, bindings)))
+    }
+
+    fun buildFor(ty: Ty, bindings: HashSet<RsPatBinding>): RsExpr {
         return when (ty) {
             is TyBool -> psiFactory.createExpression("false")
             is TyInteger -> psiFactory.createExpression("0")
             is TyFloat -> psiFactory.createExpression("0.0")
             is TyChar -> psiFactory.createExpression("''")
             is TyReference -> when (ty.referenced) {
                 is TyStr -> psiFactory.createExpression("\"\"")
-                else -> psiFactory.createRefExpr(buildFor(ty.referenced), listOf(ty.mutability))
+                else -> psiFactory.createRefExpr(buildFor(ty.referenced, bindings), listOf(ty.mutability))
             }
             is TyAdt -> {
+                val smartPointers = listOf(
+                    items.Box,
+                    items.Rc,
+                    items.Arc,
+                    items.Cell,
+                    items.RefCell,
+                    items.UnsafeCell,
+                    items.Mutex
+                )
+
                 val item = ty.item
+                if (item in smartPointers) {
+                    return buildForSmartPtr(ty, bindings)
+                }
+
+                var default = this.defaultValue
+                if (item.implLookup.isDefault(ty)) {
+                    default = psiFactory.createAssocFunctionCall("Default", "default", listOf())
```

Comment:
I don't think it's reasonable to insert `Default::default()` because it's implemented for most of the types but rarely used manually (mostly generic trait magic like `<T: default> T::default()` in some generic function).

Reply:
It seemed to me like a good default (no pun intended :slightly_smiling_face: ). I agree that it's probably not seen so often in user code, but I'd rather insert `Default::default()` than `()`, which will cause an error. So maybe use the default only as a last resort instead of `()` if possible, but otherwise prefer per-field initialization?

For me this fix would be mostly used in constructors (such as `T::new()`) and there I expect the local variable/parameter initialization will be most useful.

Reply:
Maybe it's ok... anyway I'm going to add a separate quickfix to only add missing fields without initializers #3326

Reply:
That's a good idea, personally I find the Add/Recursively add terminology a bit confusing. It could be changed to Add (just add field names and let the user fill them by moving the caret to each field successively) and Add + fill, which would work like the current recursive approach.

Reply:
So should this PR wait for #3326?